### ISSUE TITLE
Cannot pass all host config options from terraform output

### ISSUE THREAD

**Issue by jakolehm** [kontena/pharos-cluster#715]
For example `ssh_proxy_command`.

**Comment by jakolehm** [comment]
I'm quite sure we are still missing some options.

**Comment by kke** [comment]
I wonder if it would be possible to make TF output a json that would need no messing around

**Comment by jakolehm** [comment]
> I wonder if it would be possible to make TF output a json that would need no messing around

I don't know if that's possible.. TF transforms map to an array (with a map inside).

**Comment by kke** [comment]
I thinkit should at least be possible to output a json like:

```json
            "workers": [
                {
                    "address":  "127.0.0.1",
                    "private_address": "172.31.21.3",
                    "role": "worker",
                    "ssh_key_path": "~/.ssh/aws.pem",
                    "user": "ubuntu"
                },
               {
                    "address":  "127.0.0.2",
                    "private_address": "172.31.21.3",
                    "role": "worker",
                    "ssh_key_path": "~/.ssh/aws.pem",
                    "user": "ubuntu"
                }
            ]
```

Instead of the current combined one (which I think has problems in case the output has a mismatched count of private/public addresses:

```json
            "workers": [
                {
                    "address": [
                        "127.0.0.1",
                        "127.0.0.2",
                    ],
                    "private_address": [
                        "172.31.21.3",
                        "172.31.32.68",
                    ],
                    "role": "worker",
                    "ssh_key_path": "~/.ssh/aws.pem",
                    "user": "ubuntu"
                }
            ]
```

This does not yet solve the map syntax, but the `${element(..)}` looks promising for that. The syntax is super weird.

**Comment by jakolehm** [comment]
You cannot really compose that kind of output with TF, or can you?

**Comment by kke** [comment]
There's some `foreach` thing in upcoming `0.12` it seems: https://www.hashicorp.com/blog/hashicorp-terraform-0-12-preview-for-and-for-each

**Comment by kke** [comment]
`dry-type` calls `to_hash` on objects that are not hash:

```
irb(main):001:0> module ToHash
irb(main):002:1>   def to_hash
irb(main):003:2>     (size == 1 && first.is_a?(Hash)) ? first : self
irb(main):004:2>     end
irb(main):005:1>   end
=> :to_hash
irb(main):006:0> Array.include(ToHash)
=> Array
irb(main):007:0> Pharos::Configuration::Host.new(address: '1.2.3.4', bastion: [{ address: '1.2.3.4' }]).bastion
=> #<#<Class:0x00007ff917b56360> address="1.2.3.4" user=nil ssh_key_path=nil>
```

**Comment by kke** [comment]
the map/array thing can also be avoided with something like:

```
      label = "${map("ingress", "nginx")}"
```

but anything with more fields, like bastion, will become quite ugly with that.

**Comment by jakolehm** [comment]
@kke how would you pass the information from terraform to pharos with `local-exec`?

**Comment by kke** [comment]
A subcommand that would add/update the host in cluster.yml, something like:

```
resource "aws_instance" "ec2" {
        provisioner "local-exec" { 
          command = 'pharos-cluster terraform update -c cluster.yml'
          environment {
            TF_PUBLIC_IP = "${var.public_ip}"
            TF_PRIVATE_IP = "${var.private_ip}"
            ....
          }
       }
```

**Comment by jakolehm** [comment]
How about addon configs etc?

**Comment by kke** [comment]
You would have a cluster.yml without hosts like now and then terraform local-exec would modify that config to add hosts.

**Comment by jakolehm** [comment]
Fixed in #852
